main.py

```diff
@@ -1401,6 +1450,12 @@ def update_file_diagnostics(window: sublime.Window, file_path: str, source: str,
 def update_diagnostics_phantoms(view: sublime.View, diagnostics: 'List[Diagnostic]'):
     global phantom_sets_by_buffer
 
+    # disable the normal LSP phantoms when using a Rust document,
+    # since these interfere with the enhanced Rust phantoms
+    syntax = view.settings().get('syntax')
+    if "Rust" in syntax: # type: ignore
+        return
```

Comment:
This may not be the greatest solution, long-term, but maybe it'll be enough for now.

If anyone could help me understand why the phantoms are interfering with each other, maybe there's an obvious, good long-term solution to be implemented that I'm just not aware of.